src/nvim/main.c

```diff
@@ -1494,9 +1498,18 @@ static void init_params(mparm_T *paramp, int argc, char **argv)
 /// Initialize global startuptime file if "--startuptime" passed as an argument.
 static void init_startuptime(mparm_T *paramp)
 {
+  // check if we are currently running as the tui or the main process
+  // the main process is always called with the --embed flag
+  bool is_main_process = false;
+  for (int i = 1; i < paramp->argc - 1; i++) {
+    if (STRICMP(paramp->argv[i], "--embed") == 0) {
+      is_main_process = true;
+      break;
+    }
+  }
```

Comment:
This will recognize a headless instance as the TUI process.

Reply:
Meanwhile it's hard to decide whether Nvim will be launched in headless mode without parsing arguments, so maybe it's necessary to parse arguments earlier?

Reply:
good point. Can't we just explicitly search for the `--headless` option here?

```
bool is_main_process = hasFlagBeenSet("--headless") || hasFlagBeedSet("--embed");
```

Reply:
No, there are other flags that imply headless mode.

Reply:
In that case the easiest solution would probably be to rename the process in the report to something like "Primary/TUI". In that case in case of headless mode this will be the primary process, otherwise its the TUI process.

Reply:
Then the variable should be called `is_embedded_process` instead of `is_main_process`.

Reply:
yes, agree with you

Reply:
It's "embedded", not "embeded"

Reply:
oops, english is hard ;)